WireDomain/Sources/WireDomain/Notifications/Builders/Conversation/ConversationEventNotificationBuilder.swift

```diff
@@ -300,5 +156,51 @@ struct ConversationEventNotificationBuilder: NotificationBuilder {
 
         return genericMessage
     }
+}
 
+extension ConversationEventNotificationBuilder {
+    struct Validator {
+        let userLocalStore: any UserLocalStoreProtocol
+        let conversationLocalStore: any ConversationLocalStoreProtocol
+        let messageLocalStore: any MessageLocalStoreProtocol
+        
+        func validate(
+            conversationID: ConversationID,
+            senderID: UserID,
+            time: Date?
+        ) async -> Bool {
+            let conversation = await conversationLocalStore.fetchOrCreateConversation(
+                id: conversationID.uuid,
+                domain: conversationID.domain
+            )
+
+            let conversationMutedMessages = await conversationLocalStore.conversationMutedMessageTypesIncludingAvailability(
+                conversation
+            )
+
+            let isConversationMuted = conversationMutedMessages != .none
+
+            let isSelfUser = try? await userLocalStore.isSelfUser(
+                id: senderID.uuid,
+                domain: senderID.domain
+            ).isSelfUser
+
+            let eventTimeStamp = time
+            let lastReadTimestamp = await conversationLocalStore.lastReadServerTimestamp(conversation)
+
+            guard let isSelfUser,
+                    !isSelfUser,
```

Comment:
suggestion: make it easier to read
```suggestion
            guard !isSelfUser,
```

Reply:
https://github.com/wireapp/wire-ios/pull/2397/commits/b0893015cbe45b6c7970154dea7d1dacdfeed514